### ISSUE TITLE
Validation (2017): V338 (Income)

### ISSUE THREAD

**Issue by hkeeler** [cfpb/hmda-platform#157]
**Edit #:** V338
**Type:** Validity
**Category:** Loan/Application Register (only)
**Column:** Income

If applicant ethnicity, race, and sex = 4, 7, and 4, respectively, and co- applicant ethnicity, race, and sex = 5, 8, and 5, respectively, and  action taken = 1-5 or 7-8, then applicant income must = NA.

Additional resources:
- [FFIEC's 2016 HMDA File Spec](https://www.ffiec.gov/hmda/pdf/spec2016.pdf)
- [FFIEC's 2016 HMDA Edits](https://www.ffiec.gov/hmda/pdf/edit2016.pdf)
- [CFPB's 2017 HMDA File Spec](http://www.consumerfinance.gov/hmda/static/for-filers/2017/2017-HMDA-File-Specifications.pdf)

**Comment by schbetsy** [comment]
Why is this issue re-opened?

**Comment by lgdean** [comment]
Oops, I wrote it in the PR instead of here. What I said at the time was:
"I have an idea for reducing duplication in the test code, but that's no reason to keep this PR open. I'll merge it and then assign #157 to myself. If my idea works out, I'll open another PR; if not, I'll close the issue."

I still have the idea, but in the few hours of work/coding time I've had since then, I made a different PR instead.  I would like to find time to give it a try, but am open to discussion.
